scipio/src/sys/uring.rs

```diff
@@ -489,6 +489,75 @@ impl UringCommon for PollRing {
     }
 }
 
+impl InnerSource {
+    pub(crate) fn update_source_type(&mut self, source_type: SourceType) -> SourceType {
+        std::mem::replace(&mut self.source_type, source_type)
+    }
+}
+
+#[allow(clippy::mut_from_ref)]
+// This is similar to the interior mutability pattern but clippy doesn't like it.
+// That's because unlike the usual refcell patterns, we return a mutable reference
+// and the user is free to do whatever it wants with it (as opposed to a scoped RefMut).
+//
+// We don't have too much of a choice because the Source accompanies a request in
+// the io_uring and we'd be dealing with raw pointers anyway. We want to make this as
+// safe as possible but will always have to rely on those weird tricks.
+//
+// This should only really be used within the io_uring code
+fn mut_source(source: &Rc<UnsafeCell<InnerSource>>) -> &mut InnerSource {
+    unsafe { &mut *source.get() }
+}
```

Comment:
This function could be inlined in `Source::inner`, this would limit the number of individual clippy `allow` attributes to one. Otherwise the above comment should also be added to  `Source::inner` too.

Reply:
we won't be able to get rid of the two clippy markings. The function is used many other times within uring. It is not used just to implement `Source::inner`. And because it is a method of `UnsafeCell<T>` and not `T` it has to be free-floating